src/components/main/CopyButton.tsx

```diff
@@ -1,56 +1,114 @@
-import { MouseEventHandler } from 'react';
+import { useEffect, useState } from 'react';
 
 import { IconButtonProps } from '@mui/material';
 
+import { DiscriminatedItem } from '@graasp/sdk';
 import {
   ActionButtonVariant,
   CopyButton as GraaspCopyButton,
 } from '@graasp/ui';
 
+import { mutations } from '@/config/queryClient';
+import { computeButtonText } from '@/utils/itemSelection';
+
 import { useBuilderTranslation } from '../../config/i18n';
 import {
   ITEM_COPY_BUTTON_CLASS,
   ITEM_MENU_COPY_BUTTON_CLASS,
 } from '../../config/selectors';
 import { BUILDER } from '../../langs/constants';
-import { useCopyItemModalContext } from '../context/CopyItemModalContext';
+import { NavigationElement } from './itemSelectionModal/Breadcrumbs';
+import ItemSelectionModal, {
+  ItemSelectionModalProps,
+} from './itemSelectionModal/ItemSelectionModal';
 
 export type Props = {
   color?: IconButtonProps['color'];
   id?: string;
-  onClick?: MouseEventHandler<HTMLButtonElement | HTMLLIElement>;
+  onClick?: () => void;
   type?: ActionButtonVariant;
   itemIds: string[];
 };
 
 const CopyButton = ({
-  itemIds,
+  itemIds: defaultItemsIds,
   color,
   id,
   type,
   onClick,
 }: Props): JSX.Element => {
   const { t: translateBuilder } = useBuilderTranslation();
+  const { mutate: copyItems } = mutations.useCopyItems();
+  const [open, setOpen] = useState<boolean>(false);
+  const [itemIds, setItemIds] = useState<string[]>(defaultItemsIds || []);
+
+  const openCopyModal = (newItemIds: string[]) => {
+    setOpen(true);
+    setItemIds(newItemIds);
+  };
+
+  const onClose = () => {
+    setOpen(false);
+  };
 
-  const { openModal: openCopyModal } = useCopyItemModalContext();
+  const onConfirm: ItemSelectionModalProps['onConfirm'] = (payload) => {
+    // change item's root id to null
```

Comment:
Did you do that ?

Reply:
No I used the same code as in MoveModal, why ?